### PR TITLE
kubeadm: drop duplicate function NewCACertAndKey

### PR DESCRIPTION
**Pull request by rojkov**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The function certs.NewCACertAndKey() is just a wrapper around pkiutil.NewCertificateAuthority() which doesn't add any additional functionality.

Instead use pkiutil.NewCertificateAuthority() directly.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/certs/certs.go [hunk 1]
```diff
@@ -90,17 +90,6 @@ func NewServiceAccountSigningKey() (*rsa.PrivateKey, error) {
 	return saSigningKey, nil
 }
 
-// NewCACertAndKey will generate a self signed CA.
-func NewCACertAndKey(certSpec *certutil.Config) (*x509.Certificate, *rsa.PrivateKey, error) {
-
-	caCert, caKey, err := pkiutil.NewCertificateAuthority(certSpec)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "failure while generating CA certificate and key")
```
Comment (yagonobre) [inline cmd/kubeadm/app/phases/certs/certs.go]: IMO, it's provider a better error message, using only `pkiutil.NewCertificateAuthority` will return `unable to create private key` or `unable to create self-signed certificate`.
I'm not against the change, but will leave the lgtm to other reviews.

**Review by yagonobre — COMMENTED** [review]
Thanks @rojkov 
/priority backlog

File: cmd/kubeadm/app/phases/certs/certs.go (hunk 1 above)
Comment (rojkov) [reply cmd/kubeadm/app/phases/certs/certs.go]: If the sole purpose of this function is to give a better error message then the messages returned by `pkiutil.NewCertificateAuthority` could be made clearer, e.g. `unable to create private key while generating CA certificate` and `unable to create self-signed CA certificate`. What do you think?

Also additional error wrapping goes against the rationale explained in https://github.com/kubernetes/kubeadm/issues/1266

Comment (rosti) [reply cmd/kubeadm/app/phases/certs/certs.go]: You both have a point here. The func was introduced in its form by Fabrizio, so it's probably best to let him decide on this one.

/assign @fabriziopandini

**Review by fabriziopandini — CHANGES_REQUESTED** [review]
@rojkov thank for this cleanup!
please update error message returned by . pkiutil.NewCertificateAuthority as per your suggestion to address @yagonobre comment; after that ok for me

**Comment by rojkov** [comment]
Thank you for review! I've updated the error message.

**Review by yagonobre — COMMENTED** [review]
Thanks @rojkov 
/lgtm

**Review by rosti — APPROVED** [review]
Thanks @rojkov !
/approve

**Pull request merged**